Common/SecurityIdentifier.cs

```diff
@@ -404,6 +404,40 @@ public static SecurityIdentifier GenerateEquity(DateTime date, string symbol, st
             return Generate(date, symbol, SecurityType.Equity, market);
         }
 
+        /// <summary>
+        /// Generates the <see cref="Symbol"/> property for <see cref="QuantConnect.SecurityType.Base"/> security identifiers
+        /// </summary>
+        /// <param name="dataType">The base data custom data type if namespacing is required, null otherwise</param>
+        /// <param name="symbol">The ticker symbol</param>
+        /// <returns>The value used for the security identifier's <see cref="Symbol"/></returns>
+        public static string GenerateBaseSymbol(Type dataType, string symbol)
+        {
+            if (dataType == null)
+            {
+                return symbol;
+            }
+
+            return $"{symbol.ToUpperInvariant()}.{dataType.Name}";
+        }
+
+        /// <summary>
+        /// Generates a new <see cref="SecurityIdentifier"/> for a custom security
+        /// </summary>
+        /// <param name="dataType">The custom data type</param>
+        /// <param name="symbol">The ticker symbol of this security</param>
+        /// <param name="market">The security's market</param>
+        /// <returns>A new <see cref="SecurityIdentifier"/> representing the specified base security</returns>
+        public static SecurityIdentifier GenerateBase(Type dataType, string symbol, string market)
+        {
+            return Generate(
+                DefaultDate,
+                GenerateBaseSymbol(dataType, symbol),
+                SecurityType.Base,
+                market,
+                forceSymbolToUpper: false
+            );
+        }
+
         /// <summary>
         /// Generates a new <see cref="SecurityIdentifier"/> for a custom security
         /// </summary>
```

Comment:
Old `GenerateBase` is not being used anymore, should we remove it? or add some new documentation for it?

Reply:
Removed.